ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantTestContext.scala

```diff
@@ -617,16 +636,31 @@ private[testtool] final class ParticipantTestContext private[participant] (
   def submitAndWait(request: SubmitAndWaitRequest): Future[Unit] =
     services.command.submitAndWait(request).map(_ => ())
 
-  def submitAndWaitForTransactionId(request: SubmitAndWaitRequest): Future[String] =
-    services.command.submitAndWaitForTransactionId(request).map(_.transactionId)
+  def submitAndWaitForTransactionId(
+      request: SubmitAndWaitRequest
+  ): Future[SubmitAndWaitForTransactionIdResponse] =
+    services.command.submitAndWaitForTransactionId(request)
+
+  def submitAndWaitForTransaction(
+      request: SubmitAndWaitRequest
+  ): Future[SubmitAndWaitForTransactionResponse] =
+    services.command.submitAndWaitForTransaction(request)
 
-  def submitAndWaitForTransaction(request: SubmitAndWaitRequest): Future[Transaction] =
-    services.command.submitAndWaitForTransaction(request).map(_.getTransaction)
+  def submitAndWaitForTransactionReturningTransaction(
```

Comment:
I think we could mostly inline the two "returning" methods without making the code worse. Especially if we then fused the `map` calls or moved the call to `getTransaction`/`getTransactionTree` to the appropriate `yield` block

Reply:
Done, inlined everything and removed the "returning" methods.